jdbc/src/main/java/software/amazon/timestream/jdbc/TimestreamSchemasResultSet.java

```diff
@@ -15,22 +15,100 @@
 package software.amazon.timestream.jdbc;
 
 import com.amazonaws.services.timestreamquery.model.ColumnInfo;
+import com.amazonaws.services.timestreamquery.model.Datum;
+import com.amazonaws.services.timestreamquery.model.Row;
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
 import java.util.List;
 
 /**
  * Result set to return an empty list of schemas in Timestream.
  */
-public class TimestreamSchemasResultSet extends TimestreamEmptyBaseResultSet {
+public class TimestreamSchemasResultSet extends TimestreamBaseResultSet {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TimestreamDatabasesResultSet.class);
+  private static final Datum NULL_DATUM = new Datum().withNullValue(Boolean.TRUE);
   private static final List<ColumnInfo> COLUMNS = ImmutableList.of(
     TimestreamDataType.createColumnInfo(TimestreamDataType.VARCHAR, "TABLE_SCHEM"),
     TimestreamDataType.createColumnInfo(TimestreamDataType.VARCHAR, "TABLE_CATALOG"));
 
+  private boolean isAfterLast = false;
+
   /**
    * Constructor.
+   *
+   * @param connection the parent connection of the result set.
+   * @param schemaPattern the SchemaPattern. null if not provided
+   * @throws SQLException if a database access error occurs.
    */
-  public TimestreamSchemasResultSet() {
-    super(COLUMNS);
+  TimestreamSchemasResultSet(TimestreamConnection connection, String schemaPattern) throws SQLException {
+    super(null, 20);
```

Comment:
Any reason to set the fetchSize as 20? - Reference to base class code: https://github.com/awslabs/amazon-timestream-driver-jdbc/blob/4f47f779138e9a1cebf944d2c4d7de52b925a7da/jdbc/src/main/java/software/amazon/timestream/jdbc/TimestreamBaseResultSet.java#L87